### ISSUE TITLE
Assertion failure in "src/lib/reasoners/ccx.ml"

### ISSUE THREAD

**Issue by backtracking** [OCamlPro/alt-ergo#464]
On the attached file [bug.zip](https://github.com/OCamlPro/alt-ergo/files/6599769/bug.zip) Alt-Ergo 2.4.0 is raising the following error:
```
$ alt-ergo dicho-Dicho-mainqtvc.ae 
Fatal error: exception File "src/lib/reasoners/ccx.ml", line 485, characters 28-34: Assertion failed
```

**Comment by backtracking** [comment]
Note: This input file was obtained from a Why3 proof and we noticed that if we add the `eliminate_if` transformation on the Why3 side, then the problem disappears. So
* it gives you a clue on the problem, hopefully
* it might be interested to tell us (the Why3 team) whether we should always use `eliminate_if` when using Alt-Ergo 2.4.0.

**Comment by Stevendeo** [comment]
Hello,

This issue has been solved by #462, which has been merged on `next` few weeks ago. The error was indeed due to the wrong normalization of if-then-elses. I did not find any example not involving if-then-elses raising this error, so I `eliminate_if` should be the safest way to solve it. In terms of performance, I can't really say if formulas without `if-then-else` are more or less efficient (it really depends on how `eliminate_if` works).
